plugin/method-channel.go

```diff
@@ -123,55 +130,82 @@ func (m *MethodChannel) HandleFuncSync(methodName string, f func(arguments inter
 	m.HandleSync(methodName, MethodHandlerFunc(f))
 }
 
+// CatchAllHandle registers a default method handler.
+// When no Handle are found, the handler provided in CatchAllHandle will be
+// used. If no CatchAllHandle is provided, a MissingPluginException exception
+// is sent when no handler is registered for a method name.
+//
+// Consecutive calls override any existing handler registration for (the name
+// of) this method. When given nil as handler, the previously registered
+// handler for a method is unregistered.
+func (m *MethodChannel) CatchAllHandle(handler MethodHandler) {
+	m.methodsLock.Lock()
+	m.catchAllhandler = handler
+	m.methodsLock.Unlock()
+}
+
+// CatchAllHandleFunc is a shorthand for m.CatchAllHandle(MethodHandlerFunc(f))
+//
+// The argument of the function f is a MethodCall struct
+func (m *MethodChannel) CatchAllHandleFunc(f func(arguments interface{}) (reply interface{}, err error)) {
+	m.CatchAllHandle(MethodHandlerFunc(f))
+}
+
 // handleChannelMessage decodes incoming binary message to a method call, calls the
 // handler, and encodes the outgoing reply.
 func (m *MethodChannel) handleChannelMessage(binaryMessage []byte, responseSender ResponseSender) (err error) {
 	methodCall, err := m.methodCodec.DecodeMethodCall(binaryMessage)
 	if err != nil {
-		return errors.Wrap(err, "failed to decode incomming message")
+		return errors.Wrap(err, "failed to decode incoming message")
 	}
 
 	m.methodsLock.RLock()
 	registration, registrationExists := m.methods[methodCall.Method]
 	m.methodsLock.RUnlock()
 	if !registrationExists {
+
+		if m.catchAllhandler != nil {
+			go m.handleMethodCall(m.catchAllhandler, methodCall.Method, methodCall, responseSender)
```

Comment:
`(.., methodCall.Method, methodCall, ..)` 
Giving the `Method` field, and the struct containing that field? That `methodCall` should be `methodCall.Arguments`.

To me, just giving the whole methodCall instead of both fields separately feels a little bit cleaner, but I'll leave that up to you.

Reply:
Ah wait, now I get it..  This is the way the whole thing is sent as argument... Hmm..